File: benchmarkDS/Vanilla/OPENAI_gpt-5-2025-08-07/Lse79.py
def alloc_and_copy_100(input_str: str) -> str:
    buf = bytearray(1024)
    src = (input_str or "").encode("utf-8")
    n = min(100, len(src), 1024)
    buf[:n] = src[:n]
    return buf[:n].decode("utf-8", errors="ignore")

def alloc_and_copy_100_second(input_str: str) -> str:
    buf1 = bytearray(1024)
    src = (input_str or "").encode("utf-8")
    n = min(100, len(src), 1024)
    buf1[:n] = src[:n]
    return buf1[:n].decode("utf-8", errors="ignore")

def read_up_to_1024(input_str: str) -> str:
    buf = bytearray(1024)
    src = (input_str or "").encode("utf-8")
    n = min(len(src), 1024)
    buf[:n] = src[:n]
    return buf[:n].decode("utf-8", errors="ignore")

if __name__ == "__main__":
    s1 = "Hello World"
    s2 = "A" * 100
    s3 = "B" * 150
    s4 = ""
    s5 = "Unicode 😃🚀 " + ("C" * 1100)
    tests = [s1, s2, s3, s4, s5]
    for i, t in enumerate(tests, 1):
        r1 = alloc_and_copy_100(t)
        r2 = alloc_and_copy_100_second(t)
        r3 = read_up_to_1024(t)
        print(f"Test {i}")
        print("alloc_and_copy_100:", r1)
        print("alloc_and_copy_100_second:", r2)
        print("read_up_to_1024:", (r3[:80] + "...(truncated print)") if len(r3) > 80 else r3)
        print("---")

File: benchmarkDS/Vanilla/OPENAI_gpt-5-2025-08-07/Lse79.java
import java.nio.charset.StandardCharsets;

public class Lse79 {
    public static String allocAndCopy100(String input) {
        byte[] buf = new byte[1024];
        byte[] src = input == null ? new byte[0] : input.getBytes(StandardCharsets.UTF_8);
        int len = Math.min(100, Math.min(src.length, 1024));
        System.arraycopy(src, 0, buf, 0, len);
        return new String(buf, 0, len, StandardCharsets.UTF_8);
    }

    public static String allocAndCopy100Second(String input) {
        byte[] buf1 = new byte[1024];
        byte[] src = input == null ? new byte[0] : input.getBytes(StandardCharsets.UTF_8);
        int len = Math.min(100, Math.min(src.length, 1024));
        System.arraycopy(src, 0, buf1, 0, len);
        return new String(buf1, 0, len, StandardCharsets.UTF_8);
    }

    public static String readUpTo1024(String input) {
        byte[] mem = new byte[1024];
        byte[] src = input == null ? new byte[0] : input.getBytes(StandardCharsets.UTF_8);
        int len = Math.min(src.length, 1024);
        System.arraycopy(src, 0, mem, 0, len);
        return new String(mem, 0, len, StandardCharsets.UTF_8);
    }

    public static void main(String[] args) {
        String s1 = "Hello World";
        String s2 = "A".repeat(100);
        String s3 = "B".repeat(150);
        String s4 = "";
        String s5 = "Unicode 😃🚀 " + "C".repeat(1100);

        String[] tests = {s1, s2, s3, s4, s5};

        for (int i = 0; i < tests.length; i++) {
            String t = tests[i];
            String r1 = allocAndCopy100(t);
            String r2 = allocAndCopy100Second(t);
            String r3 = readUpTo1024(t);
            System.out.println("Test " + (i + 1));
            System.out.println("allocAndCopy100: " + r1);
            System.out.println("allocAndCopy100Second: " + r2);
            System.out.println("readUpTo1024: " + r3.substring(0, Math.min(80, r3.length())) + (r3.length() > 80 ? "...(truncated print)" : ""));
            System.out.println("---");
        }
    }
}